buffer.c

```diff
@@ -141,3 +151,31 @@ void editor_buffer_save(EditorBuffer* self, AdwTabView* tab_view, GtkWindow* par
     GFile* file = g_file_new_for_path(self->file_path);
     write_file(file, &self->parent);
 }
+
+AdwTabPage* editor_buffer_get_page(EditorBuffer* self){
+    return self->tab_page;
+}
+
+void set_editor_buffer_page(EditorBuffer* buffer, AdwTabPage* tab_page){
+    buffer->tab_page = tab_page;
+}
```

Comment:
This should be removed because the AdwTabPage the editor buffer is in should never change.
The Page should be set in the constructor. It also prevents the page from ever being NULL.